src/Topshelf/Runtime/Windows/WindowsServiceHost.cs

```diff
@@ -268,6 +296,10 @@ protected override void Dispose(bool disposing)
 
         void CatchUnhandledException(object sender, UnhandledExceptionEventArgs e)
         {
+            // Call the custom exception handler if it is not null
+            //
+            _settings.ExceptionCallback?.Invoke((Exception)e.ExceptionObject);

```

Comment:
Yes because the exception is provided here as an `object` type. The same technique is used to create the log statement in the unhandled exception handlers.